src/content/reference/react/useOptimistic.md

```diff
@@ -0,0 +1,212 @@
+---
+title: useOptimistic
+canary: true
+---
+
+<Canary>
+
+The `useOptimistic` Hook is currently only available in React's canary and experimental channels. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+
+</Canary>
+
+<Intro>
+
+`useOptimistic` is a React Hook that lets you optimistically update the UI.
+
+```js
+  const [optimistic, addOptimistic] = useOptimistic(initialState, updateFn);
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `useOptimistic(initialState, updateFn)` {/*use*/}
+
+`useOptimistic` is a React Hook that lets you optimistically update the UI.
+
+```js
+import { useOptimistic } from 'react';
+
+function AppContainer() {
+  const [optimistic, addOptimistic] = useOptimistic(
+    initialState,
+    (currentState, newState) => {
+      // merge and return new state
+    }
+  );
+}
+```
+
+[See more examples below.](#usage)
+
+#### Parameters {/*parameters*/}
+
+* `initialState`: `initialState` is the source of truth so if the `initialState` is ever changed and we get a new value in from the server, the return value `optimistic` will be set the that too since it will always treat `initialState` as the final source of truth.
+* `updateFn`: `updateFn` is the mutation that will occur to `optimistic` when `addOptimistic` is called.
+
+
+#### Returns {/*returns*/}
+
+* `optimistic`: `optimistic` is the optimistic state, it will default to what `initialState` is.
```

Comment:
`optimisticState`: The resulting optimistic state. It is equal to `baseState` unless an action is pending, in which case it is equal to the value returned by `reducer`.